addons/hr_fleet/wizard/hr_departure_wizard.py

```diff
@@ -12,16 +12,19 @@ class HrDepartureWizard(models.TransientModel):
     def action_register_departure(self):
         super(HrDepartureWizard, self).action_register_departure()
         if self.release_campany_car:
-            self._free_campany_car()
+            self._free_company_car()
 
-    def _free_campany_car(self):
+    def _free_company_car(self):
         """Find all fleet.vehichle.assignation.log records that link to the employee, if there is no 
         end date or end date > departure date, update the date. Also check fleet.vehicle to see if 
-        there is any record with its dirver_id to be the employee, set them to False."""
-        drivers = self.employee_id.user_id.partner_id | self.employee_id.sudo().work_contact_id
-        assignations = self.env['fleet.vehicle.assignation.log'].search([('driver_id', 'in', drivers.ids)])
-        for assignation in assignations:
-            if self.departure_date and (not assignation.date_end or assignation.date_end > self.departure_date):
-                assignation.write({'date_end': self.departure_date})
+        there is any record with its driver_id to be the employee, set them to False."""
+        drivers = self.employee_ids.user_id.partner_id | self.employee_ids.sudo().work_contact_id
+        assignations = self.env['fleet.vehicle.assignation.log'].search([
+            ('driver_id', 'in', drivers.ids),
+            '|',
+                ('date_end', '=', False),
+                ('date_end', '>', self.departure_date),
+        ])
+        assignations.write({'date_end': self.departure_date})
```

Comment:
I think it's ok because as a field **of the wizard** the `departure_date` is required?
https://github.com/odoo/odoo/blob/dda16365a83e636b9631b81109584c3d4267c79e/addons/hr/wizard/hr_departure_wizard.py#L21